back-end/services/response.service.js

```diff
@@ -0,0 +1,58 @@
+// connects to the database
+const db = require('../helpers/database');
+const Response = db.Response;
+const Question = db.Question;
+
+// Functions that we return
+module.exports = {
+    addResponse,
+    getResponsesforQuestion,
+};
+// Makes new entry into Response
+// Controllers call these functions
+async function addResponse(newResponse) {
+    console.log(`adding response: ${newResponse.answer}`);
+    // Validate request
+    if (!(newResponse.question && newResponse.answer)) {
+        throw 'Missing at least one field'
+    }
+
+    // We need to find the question id for the given question
+    let question;
+    try {
+        const questionObject = await Question.findOne({ _id: newResponse.question })
+        question = questionObject._id;
+    } catch {
+        throw `Could not find question ${newResponse.question}`
+    }
+
+    console.log(question);
+
+    if (!question) throw `Could not find question ${newResponse.question}`;
+
+    if (await Response.findOne({ question: question, answer: newResponse.answer })) {
+        throw `Response: ${newResponse.answer} for question ${newResponse.question} already exists`
+    }
+    
+    const response = { ...newResponse };
+    const dbresponse = new Response(response);
+    
+    return await dbresponse.save();
+    
+};
+
+async function getResponsesforQuestion(question) {
```

Comment:
this would aggregate for single/multiple choice, or if it's free response can just send up every response